native_client/javascript/README.md

```diff
@@ -1 +1,29 @@
 Full project description and documentation on GitHub: [https://github.com/mozilla/DeepSpeech](https://github.com/mozilla/DeepSpeech).
+
+## Generating TypeScript Type Definitions
+
+You can generate the TypeScript type declaration file using `dts-gen`.
+This requires a compiled/installed version of the DeepSpeech NodeJS client.
+
+To generate a new `index.d.ts` type declaration file, run:
+
+```sh
+npm install -g dts-gen
+dts-gen --module deepspeech --file index.d.ts
+```
```

Comment:
Since this file is now checked in, this section seems unnecessary. Future changes should be done manually to the file. For example, replacing `aSctx: object` with a proper `StreamingContext` type.

Reply:
Also, a nice future improvement would be to move the type declarations into `index.js` (making it `index.ts`), since we have TypeScript setup for `client.ts` anyway. This would also eliminate the duplication of the docs here.

Reply:
> Also, a nice future improvement would be to move the type declarations into `index.js` (making it `index.ts`), since we have TypeScript setup for `client.ts` anyway. This would also eliminate the duplication of the docs here.

That does not seems to be how people are doing it, all other packages I could check would have a `index.d.ts`.

Reply:
Yes, the final package would still have to ship index.js and index.d.ts, but `tsc` can generate a `.d.ts` file out of a `.ts` file, instead of us having to keep things in sync by hand.

Reply:
My first comment is not addressed. This section should be clarified to note that you only need to do this (regenerate `.d.ts.` with `dts-gen`) if you want to sync new changes to `index.js` into the `.d.ts`. file.